server/src/main/java/org/tbbtalent/server/service/db/impl/CandidateServiceImpl.java

```diff
@@ -1083,8 +986,8 @@ public LoginRequest register(RegisterCandidateRequest request, HttpServletReques
         }
 
         //Assign partner based on the partner abbreviation, if any
-        SourcePartner sourcePartner = (SourcePartner) partnerService.getPartnerFromAbbreviation(partnerAbbreviation);
-        if (sourcePartner == null) {
+        Partner sourcePartner = partnerService.getPartnerFromAbbreviation(partnerAbbreviation);
+        if (sourcePartner == null || !sourcePartner.isSourcePartner()) {
```

Comment:
This is a bit confusing !sourcePartner.isSourcePartner()

Reply:
Yes - I suppose so. But it is nice to have the variable named "sourcePartner" in the following code.